src/session/media_viewer.rs

```diff
@@ -0,0 +1,206 @@
+use adw::prelude::AnimationExt;
+use glib::clone;
+use gtk::prelude::*;
+use gtk::subclass::prelude::*;
+use gtk::{gdk, glib, graphene, gsk, CompositeTemplate};
+
+const ROUNDING: f32 = 12.0;
```

Comment:
That of course is a bit hacky because we don't know the border-radius of the target widget and I can't seem to find a way to retrieve it in a reasonable way, other than inspecting the entire css tree via the style context.

Reply:
There is a way: do rounding via css, animate it with a css transition :)

Reply:
The problem is that I'm doing everything in `snapshot()`, so to do that I guess I need to use a GtkPicture or smth.

Reply:
Well, yes.

Reply:
I've taken a different approach here completely. I'm doing a cross-fade between the two widgets while doing the "zoom" animation, so I don't animate the `border-radius` property anymore. It's still super WIP at the moment as there are some fixes to do to the `GtkPicture` allocation.